coalib/results/Diff.py

```diff
@@ -278,18 +276,37 @@ def modified(self):
         return result
 
     @property
+    def modified(self):
+        """
+        Calculates the modified file, after applying the Diff to the original.
+        Also adds linebreaks at the end of each line (except eof)
+        """
+        eof_linebreak = False
+
+        if self._file and self._file[-1].endswith('\n'):
+            eof_linebreak = True
+
+        file = [line.rstrip('\n') for line in self._modified()]
+
+        if eof_linebreak:
+            return self._add_linebreaks(file)
```

Comment:
Well, that's what I understood from the discussion in #3834.

If we do not need to keep the EOF-state, this change will be far less complex because we can just append linebreaks to everything.